### PR TITLE
feat(Localized Templates): ability to register user and set language,…

### PR DESCRIPTION
**Pull request by rbochenski**
… ability to have separate templates for different languages;

This is proposition to discussion:
https://github.com/directus/directus/discussions/8239

## Scope

What's changed:

- Added ability to injext user language during registrations
- Added ability to have separate templates for different languages

## Review Notes / Questions

- Right now this is just a proposition

---

Fixes #[8239]

### CODE REVIEW

**Review by DanielBiegler — CHANGES_REQUESTED** [review]
Hey there! Thank you and congratulations on your first contribution here!! :confetti_ball: :rocket: 

This will need some changes before we can merge it and if you'd like you can expand upon this idea, for example adding UI for selecting a language when [inviting](https://github.com/directus/directus/discussions/8239#discussioncomment-2898688) users.

Yesterday Kevin, Jonathan and Me did an episode on this exact topic with more infos in our Request Review Show on [DirectusTV](https://directus.io/tv/request-review) - It isnt online yet but will be soon so feel free to check that out later for more infos if youd like and theres more infos in the original discussion over [here](https://github.com/directus/directus/discussions/8239).

File: api/src/services/mail/index.ts
Comment (rbochenski) [reply api/src/services/mail/index.ts:104]: Hi, thanks for response.
I understand that yet it might happen that for different languages especially for right to left you would like to have a but different views.

So with such solution we can have both, what you propose and different templates for different languages.

Also consider situation where for some languages you might want to have different colors of buttons or so.

Proposed implementation is not conflicting with your idea, just gives additional possibilities.

Comment (rbochenski) [reply api/src/services/mail/index.ts:104]: Hi, thanks for response.
I understand that yet it might happen that for different languages especially for right to left you would like to have a but different views.

So with such solution we can have both, what you propose and different templates for different languages.

Also consider situation where for some languages you might want to have different colors of buttons or so.

Proposed implementation is not conflicting with your idea, just gives additional possibilities.

**Comment by paescuj** [comment]
Heya! Thank you for your contribution. Unfortunately, we can't accept PRs for improvements/features that don't have a prior accepted Discussion and associated Issue (as per the [Contributing Guidelines]((https://docs.directus.io/contributing/pull-request-process.html)). This is to ensure that PRs that are opened will actually get the chance to be reviewed, merged, and maintained by the core maintainers. We can reopen and continue this PR review once the requirements have been met. Thank you for your understanding!

**Pull request closed without merging**
